### PR TITLE
reading new reference position from chombo

### PR DESCRIPTION
**Pull request by rrsettgast**
Related to:
https://github.com/GEOSX/LvArray/pull/188
https://github.com/GEOSX/integratedTests/pull/76

### CODE REVIEW

File: host-configs/LBL/cori-gcc@8.1.0.cmake [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+set(ENABLE_MPI ON CACHE BOOL "")
+set(MPI_HOME /global/common/software/m3169/cori/openmpi/4.0.2/gnu/ CACHE PATH "")
+set(MPI_C_COMPILER       ${MPI_HOME}/bin/mpicc   CACHE PATH "")
+set(MPI_CXX_COMPILER     ${MPI_HOME}/bin/mpicxx  CACHE PATH "")
+set(MPI_Fortran_COMPILER ${MPI_HOME}/bin/mpifort CACHE PATH "")
+
+set(MPIEXEC              /usr/bin/srun CACHE PATH "")
+set(MPIEXEC_NUMPROC_FLAG "-n" CACHE STRING "")
+
+set(GEOSX_ENABLE_FPE OFF CACHE BOOL "" FORCE)
```
Comment (rrsettgast) [inline host-configs/LBL/cori-gcc@8.1.0.cmake:27]: @corbett5 I added a new cmake option

File: src/cmake/GeosxOptions.cmake [hunk 2]
```diff
@@ -4,6 +4,7 @@ message("CMAKE_SYSTEM_NAME = ${CMAKE_SYSTEM_NAME}")
 message("CMAKE_HOST_APPLE = ${CMAKE_HOST_APPLE}")
 
 ### OPTIONS ###
+option( GEOSX_ENABLE_FPE "" ON)
```
Comment (rrsettgast) [inline src/cmake/GeosxOptions.cmake:7]: @klevzoff @corbett5 @TotoGaz  It seems we could have more specificity to our cmake variables by prepending all of the GEOSX make variables with `GEOSX_`. Is there a reason we aren't doing this?

File: src/coreComponents/managers/initialization.cpp [hunk 3]
```diff
@@ -490,7 +492,12 @@ void finalizeLogger()
 void setupCXXUtils()
 {
   LvArray::system::setSignalHandling( []( int const signal ) { LvArray::system::stackTraceHandler( signal, true ); } );
+
+#if defined(GEOSX_USE_FPE)
```
Comment (rrsettgast) [inline src/coreComponents/managers/initialization.cpp:496]: @corbett5 This is where we use the new `GEOSX_USE_FPE` cmake variable.

File: src/cmake/GeosxOptions.cmake (hunk 2 above)
Comment (klevzoff) [reply src/cmake/GeosxOptions.cmake:7]: I'm on board with that

File: src/coreComponents/fileIO/coupling/ChomboCoupler.cpp [hunk 4]
```diff
@@ -57,35 +62,45 @@ void ChomboCoupler::write( double dt )
...
+  elemRegionManager->forElementRegionsComplete( [&]( localIndex const elemRegionIndex,
+                                                     ElementRegionBase const & elemRegion )
+  {
+    if( elemRegion.getName() == "void" )
+    {
+      voidRegionIndex = elemRegionIndex;
+    }
+  } );
 
   bool * faceMask = new bool[n_faces];
```
Comment (klevzoff) [inline src/coreComponents/fileIO/coupling/ChomboCoupler.cpp:77]: Get rid of raw `new`/`delete`?

File: src/coreComponents/fileIO/coupling/ChomboCoupler.cpp [hunk 5]
```diff
@@ -57,35 +62,45 @@ void ChomboCoupler::write( double dt )
...
 
   arrayView1d< integer const > const & ruptureState = faces->getReference< integer_array >( "ruptureState" );
-  arrayView1d< integer const > const & ghostRank = faces->getReference< integer_array >( faces->viewKeys.ghostRank );
+  arrayView1d< integer const > const & ghostRank = faces->ghostRank();
+
+  localIndex voidRegionIndex = -1;
+  elemRegionManager->forElementRegionsComplete( [&]( localIndex const elemRegionIndex,
+                                                     ElementRegionBase const & elemRegion )
+  {
+    if( elemRegion.getName() == "void" )
```
Comment (klevzoff) [inline src/coreComponents/fileIO/coupling/ChomboCoupler.cpp:71]: Is this some sort of convention for input files using Chombo to have a region named "void"?

File: src/coreComponents/fileIO/coupling/ChomboCoupler.cpp [hunk 6]
```diff
@@ -94,36 +109,79 @@ void ChomboCoupler::write( double dt )
...
-    const localIndex n_nodes = m_mesh.getNodeManager()->size();
+    const localIndex n_nodes = nodes->size();
 
     /* Build the face FieldMap. */
     FieldMap_out face_fields;
     real64 * pressure_ptr = faces->getReference< real64_array >( "ChomboPressure" ).data();
     face_fields["Pressure"] = std::make_tuple( H5T_NATIVE_DOUBLE, 1, pressure_ptr );
 
     FieldMap_out node_fields;
+    node_fields["position"] = std::make_tuple( H5T_NATIVE_DOUBLE, 3, m_referencePositionCopy.data() );
```
Comment (klevzoff) [inline src/coreComponents/fileIO/coupling/ChomboCoupler.cpp:132]: It seems that when `usePressures == false` neither pressure nor position is being read, and this function does... nothing (other than removing the file)? Just making sure if that is the intent.

File: src/cmake/GeosxOptions.cmake (hunk 2 above)
Comment (TotoGaz) [reply src/cmake/GeosxOptions.cmake:7]: That would probably reduce conflicts, so let's take it (furthermore being consistent always helps).

Comment (corbett5) [reply src/cmake/GeosxOptions.cmake:7]: I don't think we want to prefix all of the CMake variables but we definitely should prefix our macros. For instance LvArray needs to detect if OpenMP or CUDA is enabled and it is not going to look at `GEOSX_ENABLE_OPENMP`. Plus I think BLT has some logic built around common flags like `ENABLE_OPENMP` and `ENABLE_TESTS` but I could be wrong.

RAJA doesn't have a prefix on most of their options.
https://github.com/LLNL/RAJA/blob/2f99cffb2d09713030e4040ae1a8b905e6eb55eb/CMakeLists.txt#L53

File: src/coreComponents/fileIO/coupling/ChomboCoupler.cpp (hunk 4 above)
Comment (corbett5) [reply src/coreComponents/fileIO/coupling/ChomboCoupler.cpp:77]: Yeah using a `std::vector` would be much better.

File: src/coreComponents/fileIO/coupling/ChomboCoupler.cpp (hunk 6 above)
Comment (corbett5) [reply src/coreComponents/fileIO/coupling/ChomboCoupler.cpp:132]: Yeah that is the intent.

**Pull request merged**